### DISCUSSION TITLE
Install Error

### DISCUSSION THREAD

**Discussion by serghdz** [Security-Onion-Solutions/securityonion#11395] [2.4]
### Version

2.4.10

### Installation Method

Security Onion ISO image

### Description

installation

### Installation Type

Eval

### Location

on-prem with Internet access

### Hardware Specs

Meets minimum requirements

### CPU

5

### RAM

12

### Storage for /

200gb

### Storage for /nsm

0

### Network Traffic Collection

other (please provide detail below)

### Network Traffic Speeds

Less than 1Gbps

### Status

No, one or more services are failed (please provide detail below)

### Salt Status

No, there are no failures

### Logs

No, there are no additional clues

### Detail

I've tried installing seconion about 5 times now, and keep running into the same issue, i've tried giving it more storage, and cpu but the outcome is the same. I keep getting the same errors everytime.  

seconds after beginning installation, i get this first error.

![Screenshot (66)](https://github.com/Security-Onion-Solutions/securityonion/assets/84603618/c62338f1-ffb7-4d46-8eab-5a32b13b4512)

Then i get this error message

![114987239-c28a1d80-9eb6-11eb-9070-1e9d72eefbe3](https://github.com/Security-Onion-Solutions/securityonion/assets/84603618/e31942d2-caf6-47ac-ad11-ce40ba27e2f2)

I then restart, and attempt to view logs but I cannot  cd into root to view files.

I am fairly new, and currently learning. Thanks in advance!

![Screenshot (67)](https://github.com/Security-Onion-Solutions/securityonion/assets/84603618/c0eeb37e-8343-464c-a226-6433c419a433)

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to Security Onion 2.4.

**Comment by dougburks** [comment]
The first screenshot shows that it couldn't resolve the hostname `sigs.securityonion.net`. 

Have you verified that your VM has Internet access and valid DNS settings?

For more information, please see:
https://docs.securityonion.net/en/2.4/firewall.html#internet-communication

**Reply by serghdz** [reply]
I set up 2 network adapters, both host only and allowing all in promiscuous mode. Not sure how to check Dns settings, I'll look into that. Thanks for the feedback 🙏

**Reply by dougburks** [reply]
You probably want your management interface to be set to NAT or bridged so that it can access the Internet.

**Reply by serghdz** [reply]
So I'm following a course module and the guy sets both as host only adapters, and the same set up is used in several other installing tutorials I've seen and it seems to work for them. I will attempt to set one as NAT, to see if that works. Also I'll have to check my firewall rules to see if it's allowing DNS traffic.

**Reply by dougburks** [reply]
If you don't have Internet access, then you can choose our Airgap option:
https://docs.securityonion.net/en/2.4/airgap.html

However, in most cases, you'll want to figure out your network issue so that you can have Internet access and do a standard installation.

**Comment by serghdz** [comment]
I was watching the install video on security onion, and noticed that It prompts them to install the following tools, but It never prompted me this.

![seconionInstall](https://github.com/Security-Onion-Solutions/securityonion/assets/84603618/f0868c61-99bb-4978-8c6b-ea762b10cf74)


After setting the email, it goes straight to the install overview. 
![seconioninstalloverview](https://github.com/Security-Onion-Solutions/securityonion/assets/84603618/af7a45af-a5bc-49f9-9d11-ea35e359b915)


My question is, does Security Onion 2.4 automatically install them?

**Reply by dougburks** [reply]
You were viewing an old video for 2.3 and the Setup process has been streamlined for 2.4. 

We just published the new 2.4 Security Onion Essentials this morning:
https://securityonion.net/training

2.4 no longer includes Grafana or Wazuh. Osquery is included in Elastic Agent. Playbook and Strelka are still included.

**Reply by serghdz** [reply]
Hello it's me again, decided to attempt installing security onion again. I revisited the same tutorial from EC Council i'll attach pictures. They're running an older version of SO by the way.  Anyways, I follow their steps, however I keep getting the same error. 


![Screenshot (92)](https://github.com/Security-Onion-Solutions/securityonion/assets/84603618/9e6f1dbc-97f7-4fd2-a48d-66ee6db3f958)

![Screenshot (93)](https://github.com/Security-Onion-Solutions/securityonion/assets/84603618/1a3e1cfc-c23b-4d1b-8497-4ce7cae0937a)


https://github.com/Security-Onion-Solutions/securityonion/assets/84603618/cb07ad9f-b296-4961-94a9-8e58d5acd21e

(unmute to hear audio)



Now, I've watched the Seconion tutorial on the newer version. and theyre using NAT for adapter 1, my question for this approach is, from which of my adapters will i use the ip address when setting up seconion, will it be my wi-fi adapter, or the virtualbox host only adapter?  That's where I'm lost at, there's no clarification on which ip address to use in the installing Security onion pt 2 tutorial, it's hard for things to click in my head when there's no context or explanation. 

Thanks for looking

**Reply by dougburks** [reply]
> Hello it's me again, decided to attempt installing security onion again. I revisited the same tutorial from EC Council i'll attach pictures. They're running an older version of SO by the way. Anyways, I follow their steps, however I keep getting the same error.

If you have questions about EC Council training, you should probably ask EC Council. Also, if the video you shared is a clip from their paid training, they may not want it shared publicly like that.

> Now, I've watched the Seconion tutorial on the newer version. and theyre using NAT for adapter 1, my question for this approach is, from which of my adapters will i use the ip address when setting up seconion, will it be my wi-fi adapter, or the virtualbox host only adapter? That's where I'm lost at, there's no clarification on which ip address to use in the installing Security onion pt 2 tutorial, it's hard for things to click in my head when there's no context or explanation.

You shouldn't use an IP address that's already in use. You should configure Security Onion to use an unused IP address in the network range of the virtual network that it's connected to.

If you're not comfortable with virtualized networks, then you might want to consider installing on bare metal (without virtualization).

**Reply by t-myers-2024** [reply]
I am a student and have the same problem as the user above. I wish there was a tutorial on what the IPv4 address should be. I seem to have tired all different combinations, and nothing is working (NAT IPv4 address, Host Machine IPv4 address). I followed along with the YouTube videos Security Onion put out, but nothing is mentioned regarding where they came up with the IPv4 address or range of addresses. Also, the default gateway... is that supposed to be the default gateway of my local machine, or the gateway listed under the NAT device? 

It would be great to have that added section for those of us who are not seasoned but yet trying to learn.

**Reply by serghdz** [reply]
Yea, I tried at first using my ethernet adapter 2 ip addressing. However after multiple attempts and using my wifi ipv4 address I was not able to resolve it. I was able to install it using DHCP, not static, but even then lol I was unable to access the web interface. I haven't tried since, but I will this weekend. Will let you know how it goes.

**Reply by t-myers-2024** [reply]
I'm on my "100th" attempt as I type. I did do something a bit different, and it did not get caught up on the screen "could not resolve host: sigs.securityonion.net.

I chose static, and then looked to see what my NAT IPv4 address was on my local machine (in my case my NAT is named VMnet8. By the way, I did this on my local machine by going into the cmd terminal and typing in ipconfig. 

I then used an IPv4 address within that range as my static. For example, if my NAT (VMnet8) IPv4 address was 192.168.3.10... the static address I used in the Security Onion setup I used 192.168.3.20/24. 

For the gateway address I went into VMware (I know you're using something different) and clicked on "edit" within the tool bar and selected "virtual network editor." From there I clicked on NAT and then looked under the NAT settings. It shows my NAT gateway IP address there, so I used that for the gateway IP within Security Onion.

I have no clue if this will work, but if it does, I'll let you know. It takes at least an hour for everything to run through.

**Reply by t-myers-2024** [reply]
Fortunately, but unfortunately, I was able to install Security Onion successfully, but the final screen told me to visit the URL where the interface should be. There is no interface on my VM where I can access a web browser. Can we access the URL directly from the terminal?

When I rebooted my VM, it now does something that it's done in the past... a lot of "entered blocking state," "entered forwarding state," entered disabled state" regarding sobridge ports. 

So frustrating!!! I have no clue what I'm supposed to do.

**Reply by bumfighter** [reply]
If you're on the same network you just plug that SO ip address into the web browser and you should get the SOC page.

**Reply by t-myers-2024** [reply]
Thanks, but I've tried that. It gives me the "Cannot reach page error / timeout" When I say, "I've tried that" meaning I open a web browser on my host computer, because my virtual machine does not have a web browser when I install SO from their ISO file. My host machine is on the same network as my VM.

In one of the last steps regarding the install of SO it asks for something like "allowed IP address or range for web." For that, I have been putting in the range of my host computers IP address.

**Reply by bumfighter** [reply]
Can you successfully ping the SO ip from your host computer?

**Reply by AsadVathare** [reply]
I found a solution for @t-myers-2024 problem
I faced the same problem, and the solution was in configuring your firewall to allow a certain ip address for management node. 
What I did was,
sudo so-firewall includehost analyst 192.168.x.x(your ip addr in the same subnet as securityonion) 
sudo so-firewall apply

and then logged in my ubuntu vm and typed in IP address of security onion in browser and it opened.

**Comment by t-myers-2024** [comment]
Yes, I can ping both the NAT and Host-Only addresses from my Host computer. I can also successfully ping my host computer from a terminal within my VM. Just for further context, I am using VMWare to set up my virtual instances.

 

I would love to be able to get on the phone with someone at some point to try and figure this out. I’ve exhausted the number of videos and resources used to try and figure this out. Unfortunately, I do not know where else to turn. I am assuming I’m doing something wrong, or there is a firewall setting somewhere… but again, I’m a student still in the learning process. I’m much more limited in my knowledge vs. someone who has been in the industry for years.
